api/src/main/java/org/openmrs/module/oauth2login/authscheme/UsernameAuthenticationScheme.java

```diff
@@ -73,14 +73,16 @@ public Authenticated authenticate(Credentials credentials) throws ContextAuthent
 		}
 		
 		User user = getContextDAO().getUserByUsername(credentials.getClientName());
-		if (user == null) {
-			createUser(oauth2Credentials.getUserInfo());
-		} else {
-			updateUser(user, oauth2Credentials.getUserInfo());
+		if (!oauth2Credentials.isServiceAccount()) {
```

Comment:
I guess `isServiceAccount(..)` (and the boolean flag behind it) was introduced for the sole purpose of this `if` statement?

Reply:
Yes